extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java

```diff
@@ -765,6 +836,232 @@ public void close() {
 
     }
 
+    private String[] parametersAsStringArray(MethodInfo subMethod) {
+        List<Type> params = subMethod.parameters();
+        String[] result = new String[params.size()];
+        int i = 0;
+        for (Type param : params) {
+            result[i++] = param.name().toString();
+        }
+
+        return result;
+    }
+
+    private static Map<DotName, AnnotationInstance> toAnnotationMap(Collection<AnnotationInstance> annotations) {
+        Map<DotName, AnnotationInstance> result = new HashMap<>();
+        for (AnnotationInstance annotation : annotations) {
+            result.put(annotation.name(), annotation);
+        }
+        return result;
+    }
+
+    private String getHttpMethod(MethodInfo subMethod, String defaultMethod, Map<DotName, String> httpAnnotationToMethod) {
+
+        for (Map.Entry<DotName, String> annoToMethod : httpAnnotationToMethod.entrySet()) {
+            if (subMethod.annotation(annoToMethod.getKey()) != null) {
+                return annoToMethod.getValue();
+            }
+        }
+
+        for (Map.Entry<DotName, String> annoToMethod : httpAnnotationToMethod.entrySet()) {
+            if (subMethod.declaringClass().classAnnotation(annoToMethod.getKey()) != null) {
+                return annoToMethod.getValue();
+            }
+        }
+        return defaultMethod;
+    }
+
+    private void handleReturn(ClassInfo restClientInterface, String defaultMediaType, String httpMethod, String[] consumes,
+            MethodInfo jandexMethod, MethodCreator methodCreator, ResultHandle bodyValue, AssignableResultHandle builder) {
+        Type returnType = jandexMethod.returnType();
+        ReturnCategory returnCategory = ReturnCategory.BLOCKING;
+
+        String simpleReturnType = returnType.name().toString();
+        ResultHandle genericReturnType = null;
+
+        if (returnType.kind() == Type.Kind.PARAMETERIZED_TYPE) {
+
+            ParameterizedType paramType = returnType.asParameterizedType();
+            if (paramType.name().equals(COMPLETION_STAGE) || paramType.name().equals(UNI)) {
```

Comment:
At some point we should add an SPI for this, so these types are not hard coded. This is fine for now though.